lib/rules/named-spy.js

```diff
@@ -9,14 +9,33 @@ function findIdentifier (node) {
   var parent = node.parent
   while (parent) {
     if (parent.type === 'VariableDeclarator') {
-      return parent.id
+      return {
+        node: parent.id,
+        name: parent.id.name
+      }
     } else if (parent.type === 'Property') {
-      return parent.key
+      if (parent.key.type === 'Identifier') {
+        return {
+          node: parent.key,
+          name: parent.key.name
+        }
+      } else if (parent.key.type === 'Literal') {
+        return {
+          node: parent.key,
+          name: parent.key.value
+        }
```

Comment:
we'll need to add defaults to conditionals

Reply:
Not sure what you mean. If you're concerned about `key` being undefined then it doesn't matter since the `if` condition before the return checks that. Is it that? Or something else?